### PR TITLE
Retry and backoff

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: analytics.go [hunk 1]
```diff
@@ -232,11 +236,24 @@ func (c *Client) send(msgs []interface{}) {
 		return
 	}
 
+	for i := 0; i < 10; i++ {
```
Comment (f2prateek) [inline analytics.go]: 10 seems like a reasonable default? Could potentially retry it indefinitely as well.

File: analytics.go [hunk 2]
```diff
@@ -283,13 +302,13 @@ func (c *Client) loop() {
 			msgs = append(msgs, msg)
 			if len(msgs) == c.Size {
 				c.verbose("exceeded %d messages – flushing", c.Size)
-				c.send(msgs)
+				go c.send(msgs)
```
Comment (f2prateek) [inline analytics.go]: Do we need a bound on the number of goroutines this can create?

**Comment by f2prateek** [comment]
cc @tj @gjohnson

File: analytics.go (hunk 2 above)
Comment (tj) [inline analytics.go]: Hmm yeah might not be a bad idea, though hopefully we're never down enough for it to back up like mad with retries haha. Maybe just a second goro for handling retries with a high watermark that just drops after, not sure

Comment (f2prateek) [inline analytics.go]: @tj Very rough draft of what a bounded executor could look like - https://github.com/segmentio/analytics-go/commit/db3adc6760502971fa7356219fd2ba857900f20e

**Pull request merged**